coalib/settings/ConfigurationGathering.py

```diff
@@ -244,10 +245,121 @@ def load_configuration(arg_list,
+    try:
+        return TomlConfParser().parse(filename)
+    except FileNotFoundError:
+        if not silent:
+            if os.path.basename(filename) == Constants.local_coafile_toml:
+                logging.warning(COAFILE_OUTPUT
+                                .substitute(type='Local coafile',
+                                            file=Constants.local_coafile_toml,
+                                            found='not found'))
+            elif os.path.basename(filename) == '.coarc.toml':
+                logging.warning(COARC_OUTPUT
+                                .substitute(type='Requested coarc file',
+                                            file=filename,
+                                            found='does not exist'))
+            else:
+                logging.error(COAFILE_OUTPUT
+                              .substitute(type='Requested coafile',
+                                          file=filename,
+                                          found='does not exist'))
+                sys.exit(2)
+
+        return {'default': Section('default')}
+
+
+def load_toml_config(cli_sections, silent):
     if (
             bool(cli_sections['cli'].get('find_config', 'False')) and
             str(cli_sections['cli'].get('config')) == ''):
         cli_sections['cli'].add_or_create_setting(
+
+            Setting('config', PathArg(find_user_config(
+                os.getcwd(),
+                default_file='.coafile.toml'))))
+
+    # We don't want to store targets argument back to file, thus remove it
+    targets = [item.lower() for item in list(
+        cli_sections['cli'].contents.pop('targets', ''))]
+
+    if bool(cli_sections['cli'].get('no_config', 'False')):
+        sections = cli_sections
+    else:
+        base_sections = load_toml_config_file(Constants.system_coafile_toml,
+                                              silent=silent)
+        user_sections = load_toml_config_file(
+            Constants.user_coafile_toml, silent=True)
+        default_config = str(base_sections['default'].
+                             get('config', '.coafile.toml'))
+        user_config = str(user_sections['default'].
+                          get('config', default_config))
+        config = os.path.abspath(
+            str(cli_sections['cli'].get('config', user_config)))
+
+        try:
+            save = bool(cli_sections['cli'].get('save', 'False'))
+        except ValueError:
+            # A file is deposited for the save parameter, means we want to save
+            # but to a specific file.
+            save = True
+
+        coafile_sections = load_toml_config_file(config,
+                                                 silent=save or silent)
+
+        sections = merge_section_dicts(base_sections, user_sections)
+
+        sections = merge_section_dicts(sections, coafile_sections)
+
+        if 'cli' in sections:
+            logging.warning('\'cli\' is an internally reserved section name. '
```

Comment:
when a string is used more then one time, it should be a global variable in top of the file.